cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -138,22 +155,26 @@ public boolean exists(Id.Namespace namespaceId) {
    * @param metadata the {@link NamespaceMeta} for the new namespace to be created
    * @throws NamespaceAlreadyExistsException if the specified namespace already exists
    */
-  public synchronized void create(NamespaceMeta metadata)
-    throws NamespaceCannotBeCreatedException, NamespaceAlreadyExistsException {
+  @Override
+  public synchronized void create(NamespaceMeta metadata) throws Exception {
     // TODO: CDAP-1427 - This should be transactional, but we don't support transactions on files yet
     Preconditions.checkArgument(metadata != null, "Namespace metadata should not be null.");
-    Id.Namespace namespace = Id.Namespace.from(metadata.getName());
-    if (exists(Id.Namespace.from(metadata.getName()))) {
-      throw new NamespaceAlreadyExistsException(namespace);
+    NamespaceId namespace = new NamespaceId(metadata.getName());
+    if (exists(namespace.toId())) {
+      throw new NamespaceAlreadyExistsException(namespace.toId());
     }
 
+    // Namespace can be created. Check if the user is authorized now.
+    Preconditions.checkState(authorizerInstantiatorService.isRunning(), "Authorization Service has not yet started");
+    authorizerInstantiatorService.get().enforce(instanceId, getPrincipal(), Action.ADMIN);
```

Comment:
Will this throw an exception if the principal doesn't have admin access on the instanceId?